File: server/router/pas/evaluation_kpi.go
package pas

import (
	"gin-vue-admin/api/v1/pas"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitEvaluationKpiRouter(Router *gin.RouterGroup) {
	EvaluationKpiRouter := Router.Group("EvaluationKpi").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		EvaluationKpiRouter.GET("getEvaluationKpiList", pas.GetEvaluationKpiList)
		EvaluationKpiRouter.POST("setUserEvaluation", pas.SetUserEvaluation)
		EvaluationKpiRouter.POST("createEvaluationKpi", pas.CreateEvaluationKpi)   // 新建Kpi
		EvaluationKpiRouter.DELETE("removeEvaluationKpi", pas.RemoveEvaluationKpi) // 删除Kpi
		EvaluationKpiRouter.DELETE("removeEvaluationKpiByIds", pas.RemoveEvaluationKpiByIds) 
		EvaluationKpiRouter.POST("getEvaluationKpiById", pas.GetEvaluationKpiById) 
		EvaluationKpiRouter.PUT("updateEvaluationKpi", pas.UpdateEvaluationKpi) 
		EvaluationKpiRouter.GET("getLastEvaluationKpi", pas.GetLastEvaluationKpi) 
		EvaluationKpiRouter.GET("getEvaluationKpi", pas.GetEvaluationKpi) 
		
	}
}

File: web/src/api/pas/evaluationKpi.js
import service from '@/utils/request'

 export const addUserEvaluationKpi = (data) => {
    return service({
        url: "/EvaluationKpi/addUserEvaluationKpi",
        method: 'post',
        data
    })
}

export const getEvaluationKpiList = (params) => {
    return service({
        url: "/EvaluationKpi/getEvaluationKpiList",
        method: 'get',
        params
    })
}

export const setUserEvaluation= (data) => {
    return service({
        url: "/EvaluationKpi/setUserEvaluation",
        method: 'post',
        data
    })
}

export const removeEvaluationKpi = (data) => {
    return service({
        url: "/EvaluationKpi/removeEvaluationKpi",
        method: 'delete',
        data
    })
}
export const createEvaluationKpi = (data) => {
     return service({
         url: "/EvaluationKpi/createEvaluationKpi",
         method: 'post',
         data
     })
 }
 
 export const removeEvaluationKpiByIds = (data) => {
    return service({
        url: "/EvaluationKpi/removeEvaluationKpiByIds",
        method: 'delete',
        data
    })
}

export const getEvaluationKpiById = (data) => {
    return service({
        url: "/EvaluationKpi/getEvaluationKpiById",
        method: 'post',
        data
    })
}
export const updateEvaluationKpi = (data) => {
    return service({
        url: "/EvaluationKpi/updateEvaluationKpi",
        method: 'PUT',
        data
    })
}

export const getLastEvaluationKpi = (params) => {
    return service({
        url: "/EvaluationKpi/getLastEvaluationKpi",
        method: 'get',
        params
    })
}

export const getEvaluationKpi = (params) => {
    return service({
        url: "/EvaluationKpi/getEvaluationKpi",
        method: 'get',
        params
    })
}